src/V3LinkJump.cpp

```diff
@@ -83,6 +83,14 @@ class LinkJumpVisitor final : public VNVisitor {
                 underp = nodep;
                 under_and_next = false;  // IE we skip the entire while
             }
+        } else if (VN_IS(nodep, DoWhile)) {
+            // Handle it the same as AstWhile, because it will be converted to it
```

Comment:
I see that they are even the same lines as in Foreach. But there is a problem with casting, because stmtsp() methods are defined in the final classes. It is possible to add another abstract class to the hierarchy, for example AstNodeLoop, but I don't know if it's worth doing.

Reply:
Yes, we'd have to switch to `VN_CAST` and essentially duplicate the conditions. Good catch with the Foreach, if that's already done this way, I think we can keep it here as is.